### PR TITLE
Visitor cannot read restricted articles

### PR DESCRIPTION
**Pull request by LauraRe**
#### Description
_As an Owner
In order to gain subscribers
I want to be able to restrict certain content._

Changes proposed in this pull request:

* add status to articles (free/restricted)
* redirect user to subscription page if article is restricted

#### What I have learned working on this feature:
To not forget to add validation for new attributes and to test them.

### CODE REVIEW

File: features/visitor_cannot_read_restricted_articles.feature [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	As an Owner
+	In order to gain subscribers
+	I want to be able to restrict certain content.
+
+	Background:
+		Given the following articles exist
+			| title                                  | content                                                     | image                                                                 | status     | category |
+			| The Hub News is the site of the moment | Great articles! This site's popularity is raising so quick! | https://cdn.pixabay.com/photo/2016/11/29/04/19/beach-1867285__340.jpg | free       | tech     |
+			| Spring hasn't arrived in Sweden yet    | Ice can be still spotted on the street, watch out!          | https://cdn.pixabay.com/photo/2016/11/29/04/19/beach-1867285__340.jpg | restricted | sports   |
+
```
Comment (vick3d) [inline features/visitor_cannot_read_restricted_articles.feature]: delete this line

Comment (LauraRe) [reply features/visitor_cannot_read_restricted_articles.feature]: done

File: app/views/articles/index.html.haml [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
 		%p= category.name.capitalize
 		-category.articles.each do |article|
 			.article_box{id: "#{dom_id(article)}"}
-				= link_to article.title, article_path(article)
-				%p= truncate(article.content, length: 100, separator: ' ', ommision: "... (Read More)")
+				- if article.status == "free"
+					= link_to article.title, article_path(article)
+					%p= truncate(article.content, length: 100, separator: ' ', ommision: "... (Read More)")
+				- else
+					= link_to article.title, new_user_registration_path
```
Comment (tochman) [inline app/views/articles/index.html.haml]:
```
= link_to article.title,  (user_signed_in?  ?   article_path(article) : new_user_registration_path)
```

Comment (LauraRe) [reply app/views/articles/index.html.haml]: done

**Pull request merged**